pkg/controller/job/backoff_utils.go

```diff
@@ -183,12 +186,32 @@ func getFinishedTime(p *v1.Pod) time.Time {
 }
 
 func getFinishTimeFromContainers(p *v1.Pod) *time.Time {
-	var finishTime *time.Time
-	for _, containerState := range p.Status.ContainerStatuses {
-		if containerState.State.Terminated == nil {
-			return nil
+	finishTime := latestFinishTime(nil, p.Status.ContainerStatuses, nil)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.SidecarContainers) {
```

Comment:
Small non-blocking readability: I think it will be easier to follow if the Sidecar-specific logic to update the finishTime is inside the if.